nifi-commons/nifi-sensitive-property-provider/src/main/java/org/apache/nifi/properties/AbstractHashiCorpVaultSensitivePropertyProvider.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.properties;
+
+import org.apache.nifi.vault.hashicorp.HashiCorpVaultCommunicationService;
+import org.apache.nifi.vault.hashicorp.StandardHashiCorpVaultCommunicationService;
+import org.apache.nifi.vault.hashicorp.config.HashiCorpVaultProperties;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+public abstract class AbstractHashiCorpVaultSensitivePropertyProvider extends AbstractSensitivePropertyProvider {
+    private static final String VAULT_PREFIX = "vault";
+
+    private static final String URI = "vault.uri";
+    private static final String AUTH_PROPS_FILE = "vault.auth.props.file";
+    private static final String CONNECTION_TIMEOUT = "vault.connection.timeout";
+    private static final String READ_TIMEOUT = "vault.read.timeout";
+    private static final String ENABLED_TLS_CIPHER_SUITES = "vault.enabled.tls.cipher.suites";
+    private static final String ENABLED_TLS_PROTOCOLS = "vault.enabled.tls.protocols";
+    private static final String KEYSTORE = "vault.keystore";
+    private static final String KEYSTORE_TYPE = "vault.keystoreType";
+    private static final String KEYSTORE_PASSWD = "vault.keystorePasswd";
+    private static final String TRUSTSTORE = "vault.truststore";
+    private static final String TRUSTSTORE_TYPE = "vault.truststoreType";
+    private static final String TRUSTSTORE_PASSWD = "vault.truststorePasswd";
```

Comment:
It seems like declaring these property names in this class negates the value of leveraging Spring Vault's EnvironmentVaultConfiguration and the property names it expects for various authentication and access methods.  Although there is some overlap between TLS connectivity and TLS for authentication, I recommend evaluating a different approach that avoids the need to explicitly declare all of these properties.